### DISCUSSION TITLE
USB power trigger with HERO 11 can't get it to work properly

### DISCUSSION THREAD

**Discussion by MathiasMoos** [gopro/labs#649] [General]
I would like for my GOPRO HERO 11 to turn on and start recording when its get USB powered, and when the USB power is removed it should stop recording and shutdown. I have tried all kind of different ways to get it work, but simply i can't figure it out, through all my tests i have run on labs(2.20.70) but have also tried labs(2.12.70).

!MWAKE=2 / !MTUSB=1 / !uN!S!u10E
!MWAKE=2 / !MTUSB=1 / !uN!S!u10E!R
Have tried them as individually QR codes and in the same QR code and different combinations
This will do what i want it too, but when it starts recording the message "End after USB disconnect" thats all good, but right after the message it just disconnects the USB, and stops recording. For these tests i have used a USB power brick, as that is that will be used in my final setup.

I have than tried to do the same thing in GO PRO Blocky "[Visual Script Tool](https://gopro.github.io/labs/build/)", here i made the script seen in the image, this works in reverse, so when the USB is connected it stops and when its removed its starts to record, it was the only way i could get the script running. In the Blocky program i found that the "comparison""camera""USB powered" only works with the USB cable is connected to a PC and not with a USB power brick. This scripts has been run with !MWAKE=2!MTUSB=1.

"For reference, here's the image of the Blocky script and the QR code generated: 
![Image Link](https://github.com/gopro/labs/assets/93770051/9778ea3b-ea82-4898-9bf1-f2a18335fd34)"
![image](https://github.com/gopro/labs/assets/93770051/59ecc034-7e32-4731-9e81-febd8cda7fc8)

**Comment by dnewman-gpsw** [comment]
I'm questioning the quality of your power bank.  This is what I use and recommend https://www.amazon.com/Voltaic-Systems-Always-External-Battery/dp/B00XZ7YU4M  or an equivalent 'always on' power bank.  The average consumer cellphone charging power bank will turn off when insufficient current is draw. 

Testing HERO11 v2.20.70 this script worked perfectly.

![Record on USB power-11_28_2023, 08_43_18](https://github.com/gopro/labs/assets/10856084/b42b7138-def1-4c36-8277-cc6643b3f28b)

**Reply by MathiasMoos** [reply]
Okay, i have only used a USB C wall charger which i believed was good enough, but i will try to use a powerbank to see if that will do the trick

**Reply by MathiasMoos** [reply]
I have now tried it with my powerbank and a IQ charger they work, but i have my own power supply that only supply Ground and 5V in the USB C charger, and that only get the camera to turn on and than turn off again without recording. Are there some communication between the camera and power supply?

**Reply by MathiasMoos** [reply]
I have resolved the problem, it was the way i powered it, through a relay that is switched(on/off) by a raspberry pi and there was a voltage drop over the relay so the GOPRO didn't get 5V.
Thanks for your help:)

**Reply by Lin-Admin** [reply]
Does anybody have a hint what the commands in bold letters mV **4>u0**  !S <**u0**  !E+!1O!R **8** are supposed to do??
(V**4**, **<u0 ... <u0", **1O** and "R**8**")

**Reply by dnewman-gpsw** [reply]
Yes.  Command language is documented here: https://gopro.github.io/labs/control/tech/

**Comment by Lin-Admin** [comment]
I have now tried this QR code and on my Gopro 12 black it does not correctly work :-(
Starting and stopping by feeding power on USB does start and stop, but the MP4 created in DCIM shows the very first image for the whole duration of the recording.
This is independent on feeding from a strong powerbank and from a wall charger with less power. IMHO the power does not matter at all because it is only sensing presence of the voltage.

**Reply by dnewman-gpsw** [reply]
That doesn't sound like a working camera, give a factor reset and try again.  If recording of a freezed images continues, contact GoPro support.

**Reply by Lin-Admin** [reply]
Many thanks for your reply. 
I have done factory reset but this did not help. 

Two last questions:
a) How exactly contact GoPro support?
b) Would it not be time to bring the cam to seller shop for warranty replacement?

**Reply by dnewman-gpsw** [reply]
As you are struggling with Labs firmware, install stock firmware and test your camera behavior without Labs features.

a) https://community.gopro.com/s/contactus

**Reply by Lin-Admin** [reply]
@dnewman-gpsw:
Even with Labs software the non-labs functions seem ok.
Without Labs software I does not work as a Dashcam and therefore I have no use for it.
You did not answer question b)?

**Reply by dnewman-gpsw** [reply]
As the camera works with stock firmware, you have no reason to return for warranty replacement.  There is not guarantee any particular QR Code script will work with your camera, the command have changes over time and these are experimental features.  Start simple a learn and asked clear questions in your own thread, not on someone else's post.
